### ISSUE TITLE
PySide2 port

### ISSUE THREAD

**Issue by rkern** [enthought/qt_binder#59]
Currently, this is blocked by what appears to be an upstream bug. We rely on `QObject` classes to give us their `staticMetaObject` class attribute. For some reason, if you check for an existing `QApplication` instance via `QApplication.instance()` before one exists, all `QObject` classes lose their `staticMetaObject`. This is a problem, since checking for the global instance this way is standard practice.

```python
from PySide2 import QtCore, QtWidgets

print(repr(QtCore.QObject.staticMetaObject))
app = QtWidgets.QApplication.instance()
print(repr(QtCore.QObject.staticMetaObject))
if app is None:
    app = QtWidgets.QApplication([])
print(repr(QtCore.QObject.staticMetaObject))
```

**Comment by rkern** [comment]
It looks like if we look-before-we-leap using [`QApplication.startingUp()`](https://doc.qt.io/qt-5/qcoreapplication.html#startingUp), then we can avoid this. We'll have to do this in `pyface` and `traitsui` (and some other unit test suites that probably ought to be rewritten with `GuiTestAssistant`).

```python
from PySide2 import QtCore, QtWidgets

print(repr(QtCore.QObject.staticMetaObject))
if QtWidgets.QApplication.startingUp():
    app = QtWidgets.QApplication([])
else:
    app = QtWidgets.QApplication.instance()
print(repr(QtCore.QObject.staticMetaObject))
app = QtWidgets.QApplication.instance()
print(repr(QtCore.QObject.staticMetaObject))
```

**Comment by rkern** [comment]
[Reported upstream.](https://bugreports.qt.io/browse/PYSIDE-1093)

**Comment by rkern** [comment]
Fixed upstream. Waiting for upstream release.

**Comment by rkern** [comment]
Implemented in #60
